tools/cli/commands/release.js

```diff
@@ -0,0 +1,200 @@
+/**
+ * External dependencies
+ */
+import chalk from 'chalk';
+import child_process from 'child_process';
+import inquirer from 'inquirer';
+
+/**
+ * Internal dependencies
+ */
+import promptForProject from '../helpers/promptForProject';
+import { chalkJetpackGreen } from '../helpers/styling';
+import { allProjects } from '../helpers/projectHelpers';
+
+/**
+ * Command definition for the release subcommand.
+ *
+ * @param {object} yargs - The Yargs dependency.
+ * @returns {object} Yargs with the build commands defined.
+ */
+export function releaseDefine( yargs ) {
+	yargs.command(
+		'release [project] [script]',
+		'Runs a release script for a project.',
+		yarg => {
+			yarg
+				.positional( 'project', {
+					describe:
+						'Project in the form of type/name, e.g. plugins/jetpack, or type, e.g. plugins.',
+					type: 'string',
+				} )
+				.positional( 'script', {
+					alias: 's',
+					describe: 'The release script to run',
+					type: 'string',
+					choices: [ 'changelog', 'readme', 'release-branch', 'amend' ],
+				} )
+				.option( 'beta', {
+					alias: 'b',
+					describe: 'Is this a beta?',
+					type: 'boolean',
+					default: false,
+				} );
+		},
+		async argv => {
+			await releaseCli( argv );
+			if ( argv.v ) {
+				console.log( argv );
+			}
+		}
+	);
+
+	return yargs;
+}
+
+/**
+ * Handles the release command.
+ *
+ * @param {object} argv - the arguments passed.
+ */
+export async function releaseCli( argv ) {
+	// Verify we have a valid project.
+	if ( argv.project ) {
+		argv = await parseProj( argv );
+	} else {
+		argv = await promptForProject( argv );
+	}
+
+	// Check if we're working with a beta version.
+	if ( ! argv.beta ) {
+		argv = await promptBeta( argv );
+	}
+
+	// Verify we have a valid script.
+	if ( ! argv.script || argv.script === '' ) {
+		argv = await promptForScript( argv );
+	}
+
+	// Get the info we need for the script.
+	scriptRouter( argv );
+
+	// Run the script.
+	await runScript( argv );
+}
+
+/**
+ * Run the script.
+ *
+ * @param {object} argv - the arguments passed
+ */
+export async function runScript( argv ) {
+	console.log( chalkJetpackGreen( `Running ${ argv.script }! Just a moment...` ) );
+
+	try {
+		child_process.spawnSync( argv.script, [ ...argv.scriptArgs ], {
+			stdio: 'inherit',
+		} );
+	} catch ( err ) {
+		console.error( 'Error running script!', err );
+	}
+
+	console.log( argv.next );
```

Comment:
Should we only print the "next" text if the child process succeeded? i.e. both it didn't throw and it exited with a zero status code?

Reply:
Ah you're right, nice catch!